salt/kubernetes-master/init.sls

```diff
@@ -19,21 +21,31 @@ include:
 {% set extra_names = ["DNS: " + grains['fqdn'],
                       "DNS: api",
                       "DNS: api." + pillar['internal_infra_domain']] %}
+{% for extra_name in extra_names %}
+  {% do ip_addresses.append(extra_name) %}
+{% endfor %}
+
 {% for extra_name in pillar['api']['server']['extra_names'] %}
-  {% do extra_names.append("DNS: " + extra_name) %}
+  {% do ip_addresses.append("DNS: " + extra_name) %}
 {% endfor %}
 
 # add some standard extra names from the DNS domain
 {% if salt['pillar.get']('dns:domain') %}
-  {% do extra_names.append("DNS: kubernetes.default.svc." + pillar['dns']['domain']) %}
+  {% do ip_addresses.append("DNS: kubernetes.default.svc." + pillar['dns']['domain']) %}
 {% endif %}
 
 extend:
   /etc/pki/minion.crt:
     x509.certificate_managed:
-      - subjectAltName: "{{ ", ".join(extra_names + ip_addresses) }}"
-
-{% set api_ssl_port = salt['pillar.get']('api:ssl_port', '6443') %}
+      - subjectAltName: "{{ ", ".join(ip_addresses) }}"
+  /etc/kubernetes/kubelet:
+    file.managed:
+      - context:
+        schedulable: "false"
+  /etc/haproxy/haproxy.cfg:
+    file.managed:
+      - context:
+        bind_ip: "0.0.0.0"
```

Comment:
I guess this will make haproxy listen on all interfaces, right? In that case, what is the reason?